### ISSUE TITLE
Customisation via `with StoreOptions.options(...)` is broken (illustrated in "Options" tutorial)

### ISSUE THREAD

**Issue by maxalbert** [ioam/holoviews#288]
There is a bug which becomes apparent in the Options tutorial at the very [end of section "Controlling options from Python"](http://holoviews.org/Tutorials/Options.html#Controlling-options-from-Python) (last sentence). The text reads: _"Here the result is red [...]"_, but the image remains green. I can reproduce this by running the commands in an IPython notebook (version 1.3.2, git revision 8e3cedc6). Not sure if this is a bug in Holoviews itself or whether the syntax/options have changed and simply need to be updated in the tutorial.

**Comment by jlstevens** [comment]
Thanks for pointing this out, I'll be look into this problem shortly!

**Comment by jlstevens** [comment]
I've finally had a chance to look into this...sorry for the delay.

Essentially, what we wrote in the tutorial:

```python
with hv.StoreOptions.options(green_sine, 
                             options={'Image':{'style':{'cmap':'Reds'}}}):
    data, info = renderer(green_sine)
print(info)
SVG(data)
```

Does indeed reflect what happens to the local style context! At the level of ``Image``, the colormap is red as specified, made clear in this more appropriate example:

```python
img = hv.Image(np.random.rand(10,10)) # A 'fresh' object
with hv.StoreOptions.options(img,
                             options={'Image':{'style':{'cmap':'Reds'}}}):
    data, info = renderer(img)
print(info)
SVG(data)
```

The problem here is that we customized this *specific* object with the ``Greens`` colormap earlier:

```python
green_sine = image(style={'cmap':'Greens'})
```

Which adds an entry in the options system specifically for that object by its group and label (namely under the key ``'Image.Function.Sine'``) which is then a more specific declaration than the specification given to``Image``objects in general (and therefore the former takes precedence). The way to change the color to ``Reds`` for this particular object would be to use this more specific specification:

```python
with hv.StoreOptions.options(green_sine,
                             options={'Image.Function.Sine':{'style':{'cmap':'Reds'}}}):
    data, info = renderer(green_sine)
print(info)
SVG(data)
```

Once again, thanks for pointing this out!

We should have spotted this earlier - we know it can be tricky to understand the style system which is we have a tutorial in the first place. It is possible that the tutorial worked as stated at one point but once we made the style system work more consistently, we failed to update the tutorial. Hopefully, we can now make the fix and avoid confusing anyone else in future!

**Comment by jlstevens** [comment]
As this doesn't seem to be an actual bug in our code, we can fix the tutorial in two ways:

1. Try to explain the above! If people understand it, then they really will have a good grasp of the style system.
2. Create a new image object just to introduce the context manager works.

I'm not sure which approach we would prefer to take but as it is a documentation fix, it will be quick to do now. We should still try to fix it before the 1.4 release if we want to bundle notebooks in the PyPI package as we normally do.

**Comment by philippjfr** [comment]
Since it is the Options Tutorial and this needs to be explained somewhere I think we should do both.

**Comment by jlstevens** [comment]
Ok, I agree. I think we can start with showing the context manager on the new object to introduce it and then explain what is going on to style an object that has already been customized.

**Comment by jlstevens** [comment]
Fixed the tutorial in 74ce7cc. Hopefully it is cleared now!
